### PR TITLE
make tests run locally without random gems

### PR DESCRIPTION
**Pull request by grosser**
before: fun errors like this
```
rake
rake aborted!
LoadError: cannot load such file -- hoe
```

@drbrain

### CODE REVIEW

**Comment by drbrain** [comment]
I'm happy to add a Gemfile to make your development experience easier, but only as a change that is both minimal and maintainable.

The Gemfile duplicates the dependencies in the Rakefile.  Instead add the [hoe-bundler](https://github.com/flavorjones/hoe-bundler) plugin (add `Hoe.plugin :gemfile` to the Rakefile) so the Gemfile can be regenerated from the authoritative source when it gets out of date.

Use `rake bundler:gemfile` to generate a Gemfile for commit (it will have a note about which file to edit when dependencies are added or updated.)

Please remove `bundler/setup` because it is using a derivative source of dependencies, and the changes to .travis.yml.  They're not necessary as the tests pass as-is on Travis.

The Gemfile and Gemfile.lock shouldn't be shipped in the gem, so please remove them from Manifest.txt along with .bundle/config.

Also, since the Gemfile.lock should only exist at the root of the clone, please make it `/Gemfile.lock` in .gitignore, if it's popping up elsewhere in the file tree it should not be hidden.

**Comment by grosser** [comment]
- Can't remove from the manifest without breaking CI ... see https://github.com/drbrain/net-http-persistent/pull/102 ... if other bogus files are ok to ship then so is a Gemfile ?
 - It works "as is" if you know what gems it install first 🤷‍♂
 - I'd love to use a more modern release flow with an actual gemspec/Gemfile/bundler instead of this 1-off hoe stuff `rake release_to_gemcutter` <- nice

**Comment by drbrain** [comment]
> Can't remove from the manifest without breaking CI

Add an empty `install:` step to disable bundler on Travis and I think the Gemfile.lock won't appear and CI will pass with only adding `Gemfile` to Manifest.txt while I work out an update to hoe-travis and hoe-bundler.

> It works "as is" if you know what gems it install first 🤷‍♂
 
You implied in #101 that you were comfortable reading the source, but now you are not?

Then the problem you are running into is that you are looking for documentation on how to get started, so we can fix it with documentation.

Changing the official release process to one that's lacking features I use to release this gem to fix a documentation problem is unacceptable.

When the "more modern" release flow has features matching hoe and its plugins I'll consider switching.  Today it does not have those features.  You can get started by adding a patch to bundler to use Gem::PackageTask so the gem can depend on rake `file` tasks as this is a key feature I require for other projects.

**Comment by grosser** [comment]
Updated

I'm not interested in learning hoe / reading docs on it, since I'm not trying to use it in the future.

This works to add prerequisites to tasks and should work for bundler `release` task
```Ruby
task(:foo) { puts "FOO" }
task(:bar) { puts "BAR" }
Rake::Task[:foo].prerequisites << :bar
```

File: .travis.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@
 after_script:
 - rake travis:after -t
 before_script:
-- gem install hoe-travis --no-document
+- gem install hoe-travis hoe-bundler --no-document
```
Comment (drbrain) [inline .travis.yml]: Is adding this necessary?

Plugins should be installed automatically via the `travis:before` task [like this](https://travis-ci.org/drbrain/net-http-persistent/jobs/529405593#L500), can you link me to a build where this failed?

File: Rakefile [hunk 2]
```diff
@@ -1,11 +1,11 @@
 # -*- ruby -*-
 
-require 'rubygems'
 require 'hoe'
 
 Hoe.plugin :git
 Hoe.plugin :minitest
 Hoe.plugin :travis
+Hoe.plugin :bundler
```
Comment (drbrain) [inline Rakefile]: These should be an alphabetic order

File: Rakefile [hunk 3]
```diff
@@ -22,6 +22,9 @@ Hoe.spec 'net-http-persistent' do
 
   dependency 'connection_pool',   '~> 2.2'
   dependency 'minitest',          '~> 5.2', :development
+  dependency 'minitest',          '~> 5.2', :development
```
Comment (drbrain) [inline Rakefile]: This dependency is duplicated

File: Rakefile [hunk 4]
```diff
@@ -22,6 +22,9 @@ Hoe.spec 'net-http-persistent' do
...
+  dependency 'hoe-travis',        '>= 1.3', :development
```
Comment (drbrain) [inline Rakefile]: hoe-travis must be version 1.4.1 or newer, and less than 2.x so the correct requirement is `'~> 1.4', '>= 1.4.1'`

File: Rakefile [hunk 5]
```diff
@@ -22,6 +22,9 @@ Hoe.spec 'net-http-persistent' do
...
+  dependency 'hoe-bundler',       '>= 1.5', :development
```
Comment (drbrain) [inline Rakefile]: Use "~>" for the requirement as a 2.x version is expected to be incompatible.

**Comment by drbrain** [comment]
> This works to add prerequisites to tasks and should work for bundler release task

Providing me with workarounds for a defective tool is not convincing me to use it

**Comment by grosser** [comment]
it's not a workaround, it's how rake tasks are used 🤷‍♂ 
same as `task foo: [:bar]`

**Comment by grosser** [comment]
alls clean now I hope

**Comment by grosser** [comment]
@drbrain good to go ?

**Comment by drbrain** [comment]
PS: I might know a little about rake tasks and how best to use them as I have been rake maintainer.  Bundler is using them incorrectly and I won't consider using it until it's fixed.  Stop trying to argue with me on this, it's not endearing.

**Comment by grosser** [comment]
I did not know you have a backstory with rake ... I used prerequisites a lot and it was stable so far 🤷‍♂

**Pull request merged**